clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -186,6 +193,15 @@ protected void handleFetchSuccess(final Node fetchTarget,
                 log.debug("Fetch {} at offset {} for partition {} returned fetch data {}",
                         fetchConfig.isolationLevel, fetchOffset, partition, partitionData);
 
+                Errors partitionError = Errors.forCode(partitionData.errorCode());
+                if (requestVersion >= 16 && (partitionError == Errors.NOT_LEADER_OR_FOLLOWER || partitionError == Errors.FENCED_LEADER_EPOCH)) {
```

Comment:
Do we want to make `16` a static final, since it's used below too?

Reply:
@ex172000 That commit is out-dated, that check against `requestVersion` is not needed, has been removed since then.